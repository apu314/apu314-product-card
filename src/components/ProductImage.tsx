import type { CSSProperties, FC, SyntheticEvent } from "react"

import React, { useContext } from "react"

import { ProductContext } from "./ProductCard"

import noImage from '../assets/no-image.jpg'

import styles from '../styles/styles.module.css'


export interface Props {
  img?: string
  alt?: string
  className?: string
  style?: CSSProperties
}

export const ProductImage: FC<Props> = ({ img = '', alt = '', className, style }) => {

  const { product } = useContext(ProductContext)
  let imgToShow: string

  if (img) {
    imgToShow = img
  } else if (product.img) {
    imgToShow = product.img
  } else {
    imgToShow = noImage
  }

  const onErrorImage = (e: SyntheticEvent<HTMLImageElement, Event>) => e.currentTarget.src = noImage

  return (
    <img
      className={`${styles.productImg} ${className}`}
      style={style}
      src={imgToShow}
      onError={onErrorImage}
      alt={alt}
    />
  )
}
