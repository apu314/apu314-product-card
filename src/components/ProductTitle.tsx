import type { CSSProperties, FC } from "react"

import React, { useContext } from "react"

import { ProductContext } from "./ProductCard"

import styles from '../styles/styles.module.css'


export interface Props {
  title?: string
  className?: string
  style?: CSSProperties
}

export const ProductTitle: FC<Props> = ({ title = '', className, style }) => {
  const { product } = useContext(ProductContext)

  return (
    <span
      className={`${styles.productDescription} ${className}`}
      style={style}
    >
      {title ? title : product.title}
    </span>
  )
}
