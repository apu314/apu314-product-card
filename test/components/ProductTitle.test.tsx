import React from 'react';
import renderer from 'react-test-renderer';
import { ProductCard, ProductTitle } from '../../src/components';
import { product1 } from '../data/products';

describe('ProductTitle', () => {
  test('should show the title component with a custom title', () => {
    const wrapper = renderer.create(<ProductTitle title="Custom Product" />);

    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('should show component with product name', () => {
    const wrapper = renderer.create(
      <ProductCard product={product1}>
        {
          () => (
            <ProductTitle />
          )
        }
      </ProductCard>
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
