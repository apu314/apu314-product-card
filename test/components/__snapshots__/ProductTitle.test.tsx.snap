// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTitle should show component with product name 1`] = `
<div
  className="productCard undefined"
>
  <span
    className="productDescription undefined"
  >
    Coffee Mug - Card
  </span>
</div>
`;

exports[`ProductTitle should show the title component with a custom title 1`] = `
<span
  className="productDescription undefined"
>
  Custom Product
</span>
`;
